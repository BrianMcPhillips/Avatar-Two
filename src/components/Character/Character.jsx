import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Character = ({ name, image, id }) => (
  <figure>
    <Link to={`/detail/${id}`}>
      <img src={image} alt={name} />
      <figcaption>{name}</figcaption>
    </Link>
  </figure>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Character;
