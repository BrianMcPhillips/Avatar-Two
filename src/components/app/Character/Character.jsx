import React from 'react';
import PropTypes from 'prop-types';

const Character = () => (

);

Character.propTypes = {

};

export default Character;
