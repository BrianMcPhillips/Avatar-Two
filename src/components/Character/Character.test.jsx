import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';
import { MemoryRouter } from 'react-router-dom';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Character 
          name="Test"
          image="test.png"
          id={1234}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
