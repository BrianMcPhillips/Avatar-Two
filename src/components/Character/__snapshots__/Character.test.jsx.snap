// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <figure>
    <a
      href="/detail/1234"
    >
      <img
        alt="Test"
        src="test.png"
      />
      <figcaption>
        Test
      </figcaption>
    </a>
  </figure>
</DocumentFragment>
`;
